### DISCUSSION TITLE
Support of Enterprise Program API

### DISCUSSION THREAD

**Discussion by mkj-is** [fastlane/fastlane#22205] [Ideas] [answered]
On Aug 16 Apple published the Enterprise Program API with OpenAPI spec to manage users, roles, provisioning profiles and bundle identifiers. This is most definitely worth exploring.

- https://developer.apple.com/documentation/enterpriseprogramapi
- https://developer.apple.com/news/site-updates/?id=08162024a

First it was briefly mentioned in ["What's new in App Store Connect" WWDC24 video](https://developer.apple.com/wwdc24/10063).

**Comment by GevaZeichner** [comment]
Also talked about briefly [here](https://github.com/fastlane/fastlane/discussions/18049#discussioncomment-10419416)
I experimented with it today, hard-coding the new endpoint since it closely matches the Connect API. I got Spaceship working and successfully retrieved Enterprise provisioning data (certificates, profiles, bundle IDs) using an enterprise key. Next, I’m planning to explore what it takes to support a broader workflow.

**Reply by Westacular** [reply]
There's an existing `in_house` flag in Fastlane's `Token` class that's used to represent the Connect API keys. (Never understood why that existed, since Enterprise keys weren't available until now...)

You could probably use that flag to dynamically select which endpoint to use?

**Reply by GevaZeichner** [reply]
I tested this today with `cert`, and it’s working well. I’ll go ahead and add the dynamic endpoint selection, which might be all that’s needed. 🤞

**Reply by AlexTrott** [reply]
Haven't been able to validate yet, but do we think this is all that's needed? https://github.com/fastlane/fastlane/compare/master...AlexTrott:fastlane:feature/enterprise-program-api

**Reply by GevaZeichner** [reply]
This seems pretty much all is needed. I already validated the solution with Spaceship & Cert (retrieving data and creating a new certificate). Here's the PR: #22215 🎉 
Working on validating it further in Match and Sigh.

**Reply by AStrickland-abbott** [reply]
Any plans on getting the PR merged? Still in testing phase?

**Reply by GevaZeichner** [reply]
I finished my testing, and the PR was approved. I asked in the PR now if anything else is needed for it to be merged.

**Reply by GevaZeichner** [reply, marked as answer]
Merged!

**Reply by AStrickland-abbott** [reply]
Do we need to update any documentation about the new support?

**Reply by AlexTrott** [reply]
feels like this 100%

> Note: The App Store Connect API does not yet have the ability to determine if the team is App Store or Enterprise. The app_store_connect_api_key action and the fastlane API Key JSON file format allow for an optional in_house key as a temporary workaround

Reference: 
https://docs.fastlane.tools/app-store-connect-api/


and maybe the table above to cover what apple do support for enterprise program api
